From d3a3d233179c1834e78b7a355f810ee9789f6d78 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:40 -0400
Subject: [iommu] Change iommu driver to call detach_device_to_domain trace event

Message-id: <20140810182040.24055.38603.stgit@gir.stowe>
Patchwork-id: 87583
O-Subject: [RHEL7.1 PATCH 12/19] iommu: Change iommu driver to call detach_device_to_domain trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 699806302d8ac7dfb7d46e20da0ecd4317418d1d

commit 699806302d8ac7dfb7d46e20da0ecd4317418d1d
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:27 2013 -0600

    iommu: Change iommu driver to call detach_device_to_domain trace event

    Change iommu driver to call detach_device_to_domain trace event. This
    iommu_device class event can be enabled to trigger when devices are detached
    from a domain. Trace information includes device name.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

           swapper/0-1     [003] ....     1.854102: detach_device_from_domain: I

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index 74c371c..b630754 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -702,6 +702,7 @@ void iommu_detach_device(struct iommu_domain *domain, struct device *dev)
   return;
 
  domain->ops->detach_dev(domain, dev);
+ trace_detach_device_from_domain(dev);
 }
 EXPORT_SYMBOL_GPL(iommu_detach_device);
 
-- 
1.7.1